src/main/java/org/openmrs/addonindex/scheduled/FetchDetailsToIndex.java

```diff
@@ -147,7 +148,8 @@ void fetchExtraDetailsForEachVersion(AddOnToIndex toIndex, AddOnInfoAndVersions
 					if (configXml == null) {
 						throw new IllegalArgumentException("No config.xml file in " + version.getDownloadUri());
 					} else {
-						handleConfigXml(configXml, version);
+						handleConfigXml(configXml, version, infoAndVersions, idAndPackageSet);
+						idAndPackageSet = true;
```

Comment:
I added the boolean just because the iteration is already sorted and the latest version is fetched first